lib/vfscore/lookup.c

```diff
@@ -223,21 +227,37 @@ namei(const char *path, struct dentry **dpp)
 				node[0] = 0;
 
 				if (++links_followed >= MAXSYMLINKS) {
-					return (ELOOP);
+					error = ELOOP;
+					goto out;
 				}
 				need_continue = 1;
 				break;
 			}
 
 			if (*p == '/' && ddp->d_vnode->v_type != VDIR) {
 				drele(ddp);
-				return ENOTDIR;
+				error = ENOTDIR;
+				goto out;
 			}
 		}
 	} while (need_continue);
 
 	*dpp = dp;
-	return 0;
+out:
+	if (realpath)
+		strcpy(realpath, fp);
+	return error;
+}
+/*
+ * Convert a pathname into a pointer to a dentry
```

Comment:
I think the commentary is not needed, since it's very similar in [vfs.h](https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/vfs.h#L675)

Reply:
Indeed the comments are duplicates of the docstrings in vfs.h, however the file already has this style for exported functions and I don't want to break it without good reason.